pulsar-jms-tracing/src/main/java/com/datastax/oss/pulsar/jms/tracing/BrokerTracing.java

```diff
@@ -0,0 +1,484 @@
+                      pulsar
+                          .getConfiguration()
+                          .getProperties()
+                          .getProperty("jmsTracingLevel", defaultTraceLevel.toString());
+                  try {
+                    return TraceLevel.valueOf(level);
+                  } catch (IllegalArgumentException e) {
+                    log.warn(
+                        "Invalid tracing level: {}. Using default: {}", level, defaultTraceLevel);
+                    return defaultTraceLevel;
+                  }
+                }
+              });
+
+  private static final LoadingCache<Subscription, TraceLevel> traceLevelForSubscription =
+      CacheBuilder.newBuilder()
+          .expireAfterWrite(10, TimeUnit.SECONDS)
+          .build(
+              new CacheLoader<Subscription, TraceLevel>() {
+                public TraceLevel load(Subscription sub) {
+                  if (sub.getSubscriptionProperties() == null
+                      || !sub.getSubscriptionProperties().containsKey("trace")) {
+                    return TraceLevel.NONE;
+                  }
+
+                  try {
+                    return TraceLevel.valueOf(sub.getSubscriptionProperties().get("trace"));
+                  } catch (IllegalArgumentException e) {
+                    log.warn(
+                        "Invalid tracing level: {}. Setting to NONE for subscription {}",
+                        sub.getSubscriptionProperties().get("trace"),
+                        sub);
+                    return TraceLevel.NONE;
+                  }
+                }
+              });
+
+  private static final LoadingCache<Producer, TraceLevel> traceLevelForProducer =
+      CacheBuilder.newBuilder()
+          .expireAfterWrite(10, TimeUnit.SECONDS)
+          .build(
+              new CacheLoader<Producer, TraceLevel>() {
+                public TraceLevel load(Producer producer) {
+                  if (producer.getMetadata() == null
+                      || !producer.getMetadata().containsKey("trace")) {
+                    return TraceLevel.NONE;
+                  }
+                  try {
+                    return TraceLevel.valueOf(producer.getMetadata().get("trace"));
+                  } catch (IllegalArgumentException e) {
+                    log.warn(
+                        "Invalid tracing level: {}. Setting to NONE for producer {}",
+                        producer.getMetadata().get("trace"),
+                        producer);
+                    return TraceLevel.NONE;
+                  }
+                }
+              });
+
+  public void initialize(PulsarService pulsarService) {
+    log.info("Initializing BrokerTracing");
+  }
+
+  @Override
+  public void close() {
+    log.info("Closing BrokerTracing");
+  }
+
+  private Set<EventReasons> getEnabledEvents(ServerCnx cnx) {
+    if (cnx == null) return new HashSet<>();
+
+    return getEnabledEvents(cnx.getBrokerService().getPulsar());
```

Comment:
why can't we cache the value in "initialize" ?

Reply:
@eolivelli I think the config is dynamically updated (DynamicConfiguration something), re-reading value avoids need for broker restart